node/packages/client/src/ethereum/parser/foxy.ts

```diff
@@ -0,0 +1,69 @@
+import { Tx as BlockbookTx } from '@shapeshiftoss/blockbook'
+import { ethers } from 'ethers'
+import { TxParser } from '../../types'
+import { TxSpecific, SubParser } from '../types'
+import { getSigHash, txInteractsWithContract } from './utils'
+import { FOXY_STAKING_CONTRACT } from './constants'
+import FOXY_STAKING_ABI from './abi/foxyStaking'
+
+export class Parser implements SubParser {
+  abiInterface: ethers.utils.Interface
+
+  readonly stakeSigHash: string
+  readonly unstakeSigHash: string
+  readonly instantUnstakeSigHash: string
+  readonly claimWithdrawSigHash: string
+
+  constructor() {
```

Comment:
As this class accepts no arguments, do we actually need a constructor?

Reply:
In order to use the initialized `sigHash` properties, I believe you still need a constructor.

Reply:
With the current implementation, yes - see https://github.com/shapeshift/unchained/pull/387/files/29de97a59c4f169870023a013d793e8ec000d83a#r852436841 for an alternative to consider.